### PR TITLE
[CognitoIdentityProvider] ForgotPassword

### PR DESCRIPTION
**Pull request by rmikalkenas**

### CODE REVIEW

**Comment by jderusse** [comment]
Hello thanks for your contribution. I'm curious, what's your use case ?
You can create one  PR for several operation.

**Comment by rmikalkenas** [comment]
Hi! :) I would like to replace base aws sdk library with async one, but its missing some methods I use on my project. So all these 6 pull requests will cover my needs (SignUp, InitateAuth, RespondToAuthChallenge, ResendConfirmationCode, ForgotPassword and ConfirmForgotPassword)

File: src/Service/CognitoIdentityProvider/tests/Unit/Result/ForgotPasswordResponseTest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        ]));
+
+        $client = new MockHttpClient($response);
+        $result = new ForgotPasswordResponse(new Response($client->request('POST', 'http://localhost'), $client, new NullLogger()));
+
+        self::assertEquals(
+            CodeDeliveryDetailsType::create(
+                [
+                    'AttributeName' => 'a',
+                    'DeliveryMedium' => 'b',
```
Comment (jderusse) [inline src/Service/CognitoIdentityProvider/tests/Unit/Result/ForgotPasswordResponseTest.php:33]: Could you please use real Enum ?

Comment (rmikalkenas) [reply src/Service/CognitoIdentityProvider/tests/Unit/Result/ForgotPasswordResponseTest.php]: looks ok now

**Comment by jderusse** [comment]
Thanks you @rmikalkenas

**Pull request merged**